### PR TITLE
Rewrite icon action test in user controller

### PR DESCRIPTION
**Pull request by eduardoj**
Fixes: #5303

Co-authored-by: Dany Marcoux <dmarcoux@suse.com>

### CODE REVIEW

**Comment by eduardoj** [comment]
Do not merge, at the test is not finished yet.

File: src/api/spec/controllers/webui/user_controller_spec.rb [hunk 1]
```diff
@@ -340,20 +340,26 @@
...
-      image = ChunkyPNG::Image.from_blob(response.body)
-      expect(image.height).to be 20
-      expect(image.width).to be 20
+    context 'with a user not having an icon' do
+      let(:user) { instance_double('User', login: 'test_user') }
+
+      before do
+        get :icon, params: { user: user.login, size: 20 }
+
+        allow(user).to receive(:gravatar_image).with(20).and_return(:none)
```
Comment (coolo) [inline src/api/spec/controllers/webui/user_controller_spec.rb]: I find this a very bad test as it basically disables the main function that this route is using. Better mock the actual gravatar calls if they are troublesome in disconnected tests. But for a given email address, we expect a given gravatar URL to be loaded.

File: src/api/spec/controllers/webui/user_controller_spec.rb [hunk 2]
```diff
@@ -340,20 +340,26 @@
...
+      end
+
+      it { expect(response.location).to match(/.*\/assets\/default_face-.*\.png$/) }
```
Comment (coolo) [inline src/api/spec/controllers/webui/user_controller_spec.rb]: what you don't test at all anymore is that the route does something proper with the size param. It's just a matter of time that a future refactoring will make sure we load wallpaper sized icons into an

**Comment by eduardoj** [comment]
The purpose of this pull request is testing the `icon` action, and only test there what the action does. It will be needed also to add tests for the `gravatar_image` method in the `user` model.

**Comment by coolo** [comment]
but you only test the action under the assumption that gravatar_image returns nil - that's basically a third of the action's code

**Comment by dmarcoux** [comment]
@coolo It's WIP. The different cases will be covered.

**Comment by eduardoj** [comment]
@coolo:

> but you only test the action under the assumption that gravatar_image returns nil - that's basically a third of the action's code

As the root of the problem was found (not following a redirection), and this case is already handled in this `icon` action tests, and makes the flickering tests stop flickering, we will add the tests for the `gravatar_image` method model in another pull request.

Removing 'DO NOT MERGE' label.

**Comment by bgeuken** [comment]
I would rather replace this action with a partial or helper. In the end this is just an image we want to show within a page, or not?

**Comment by coolo** [comment]
Well, the usual story: when this started, few people had a gravatar image and quick access to the default face was important, so we cached the fact someone had no gravatar image and served the default face.

Now a lot of people have gravatars and OBS internal caching makes little sense. This would drop a lot of code

**Comment by hennevogel** [comment]
So let's drop this :-)

**Comment by coolo** [comment]
may I? :)

**Comment by coolo** [comment]
https://github.com/openSUSE/open-build-service/pull/5352 - it's only tested manually

**Pull request merged**